test/lib/commands/publish.js

```diff
@@ -889,6 +889,15 @@ t.test('latest dist tag', (t) => {
     }, new Error('Cannot implicitly apply the "latest" tag because published version 100.0.0 is higher than the new version 99.0.0. You must specify a tag using --tag.'))
   })
 
+  t.test('PREVENTS publish when latest version is SAME AS publishing version', async t => {
+    const version = '100.0.0'
+    const { npm, registry } = await loadNpmWithRegistry(t, init(version))
+    registry.publish(pkg, { noPut: true, packuments })
+    await t.rejects(async () => {
+      await npm.exec('publish', [])
+    }, new Error('You cannot publish over the previously published versions: 100.0.0.'))
+  })
```

Comment:
Optional: With the new error check, you could add additional tests unrelated to the `latest` tag. Maybe `100.0.0` has the `next` tag, or was published and then unpublished.

Reply:
test added!